jetty-server/src/main/java/org/eclipse/jetty/server/ProxyConnectionFactory.java

```diff
@@ -662,6 +662,10 @@ private void parseBodyAndUpgrade() throws IOException
                     if (LOG.isDebugEnabled())
                         LOG.debug("Proxy v2 {} {}", getEndPoint(), proxyEndPoint.toString());
                 }
+                else
+                {
+                    _buffer.position(_buffer.position() + Math.min(_buffer.remaining(), _length));
```

Comment:
I don't think this is correct.
By this point, the buffer is guaranteed to have at least `_length` bytes in it, so `Math.min()` is not necessary.
Furthermore, I think it's wrong for LOCAL to upgrade the connection (it happens few lines below) - it was probably wrong before too.
I don't know exactly what needs to be done when receiving a LOCAL command though - perhaps just read and discard until -1. Let's discuss this.

Reply:
Regarding upgrading: as discussed; the connection should be upgraded and handled normally even when the command is LOCAL.

Regarding `Math.min()`, you're right that it's not needed as the buffer is guaranteed to contain at least `_length` bytes at that point. So I'll remove it.